src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -0,0 +1,153 @@
+    visited.add(indexAsDataArray);
+    let visitSize: number = 1;
+
+    // Loop until no more adjacent pixel's within tolerance level
+    while(visitSize != 0) {
+      let coord: Array<number> = visit.pop();
+      visitSize -= 1;
+      // Unpack coord
+      let x: number = coord[0];
+      let y: number = coord[1];
+
+      // Operational part of while-loop
+      mask.add(this.coordToDataArrayIndices(x, y, imgData.width)[0]);
+
+      // Loop part of while-loop
+
+      // Get coords of adjacent pixels
+      let leftPixelCoord: Array<number> = [x - 1, y];
+      let rightPixelCoord: Array<number> = [x + 1, y];
+      let upPixelCoord: Array<number> = [x, y - 1];
+      let downPixelCoord: Array<number> = [x, y + 1];
+      // Add coords of adjacent pixels to the heap
+      let isStillMask: boolean = this.getIsStillMask(originalPixel, imgData, leftPixelCoord, tolerance, visited);
+      // Check left pixel
+      if (isStillMask) {
+        visit.push(leftPixelCoord);
+        visitSize += 1;
+      }
+      isStillMask = this.getIsStillMask(originalPixel, imgData, rightPixelCoord, tolerance, visited);
+      // Check right pixel
+      if (isStillMask) {
+        visit.push(rightPixelCoord);
+        visitSize += 1;
+      }
+      isStillMask = this.getIsStillMask(originalPixel, imgData, upPixelCoord, tolerance, visited);
+      // Check up pixel
+      if (isStillMask) {
+        visit.push(upPixelCoord);
+        visitSize += 1;
+      }
+      isStillMask = this.getIsStillMask(originalPixel, imgData, downPixelCoord, tolerance, visited);
+      // Check down pixel
+      if (isStillMask) {
+        visit.push(downPixelCoord);
+        visitSize += 1;
+      }
+    }  // end of while loop
+
+    return mask;
+  }
+
+  /** Judge current pixel's RGBA against original pixel's RGBA to see if it
+   *  can still be part of the mask (using tolerance criteria)
+   */
+  getIsStillMask(originalPixel: Array<number>, imgData: ImageData,
+                         pixelCoord: Array<number>, tolerance: number,
+                         visited: Set<number>): boolean {
+    let isStillMask: boolean = true;
+
+    // Preface; check if pixel is valid (indexing errs and repeat values)
+    let isValid: boolean = this.getIsValid(imgData.width, imgData.height, pixelCoord, visited);
+    if (!isValid) {
+      // Automatically not in mask b/c failed vailidity test
+      return !isStillMask;
+    }
+
+    // Get array of attributes of current pixel
+    let curX: number = pixelCoord[0];
+    let curY: number = pixelCoord[1];
+    let curPixel: Array<number> = this.dataArrayToRGBA(imgData, curX, curY);
+
+    // All attributes of the pixel (R,G,B, and A) must be within tolerance level
+    for (let i = 0; i < 4; i++) {
+      let upperTolerance: boolean = curPixel[i] > originalPixel[i] + tolerance;
+      let lowerTolerance: boolean = curPixel[i] < originalPixel[i] - tolerance;
+      let failsTolerance: boolean = upperTolerance || lowerTolerance;
```

Comment:
no need to introduce an additional variable for this.  It requires the code reader to have to remember what condition this satisfies when inside the if() statement.  Since it isn't a complicated expression I would just put it directly in.